scavenger-collector/src/main/kotlin/com/navercorp/scavenger/repository/sql/JvmSql.kt

```diff
@@ -55,4 +55,17 @@ class JvmSql : SqlGeneratorSupport() {
             customerId = :customerId
             AND uuid IN ( :uuids )
         """.trimIndent()
+
+    fun deleteAllByWithoutAgent(): String =
```

Comment:
@tngusdl3719 Scavenger is also supported by Vitess. Vitess does not support subqueries in sharded DML.
Also, we're using a shard key set to customerId.
How do you like these ways?

https://github.com/naver/scavenger/blob/d46ff3eb5d876e973736855e13a6f23d3acf03c7/scavenger-collector/src/main/kotlin/com/navercorp/scavenger/service/GarbageCollectService.kt#L97-L113


Also, we're using a shard key set to customerId.
How do you like these ways?
Before executing 'jvmDao.deleteAllByCustomerIdAndUuids', execute the following query.

- query the JVM, which does not exist in the agent with a shard key.
```sql
SELECT uuid FROM
    jvms
WHERE
    customerId = :customerId
    AND uuid NOT IN (
        SELECT
            jvmUuid
        FROM
            agent_state
    );
```

execute `jvmDao.deleteAllByCustomerIdAndUuids` with the result of the query.
```kotlin
jvmDao.deleteAllByCustomerIdAndUuids(customerId, agentStates.map { it.jvmUuid } + results)
```

Reply:
I overlooked some aspects of Vitess and sharding.

I have made changes based on the feedback you provided.

thx